### PR TITLE
Make list of docs from README

### PR DESCRIPTION
**Pull request by peterbrightwell**
Default for new repos should be to use docs/README.md to specify the doc list order and levels, rather than use document numbering.

### CODE REVIEW

**Comment by garethsb** [comment]
Would it be worth making clear in the comment that bullets have to be `-` and any other constraints on link line syntax?
Is it clear that all documents have to be listed here, i.e. not OK to have some that are just linked from other docs?
Are all links relative to _docs/_?
Would it be alright if people organised files in sub-directories?

File: docs/README.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
-This directory is for Markdown documentation.
+<!-- 
+Use this file to provide a index to the documents.
```
Comment (garethsb) [inline docs/README.md]:
```suggestion
Use this file to provide an index to the documents.
```

:-)

Comment (peterbrightwell) [reply docs/README.md]: @garethsb I've added a lot more instructions in the comment.

**Review by garethsb — APPROVED** [review]
Perfect! The bit about level 3 headings was the other thing I meant to mention. Thanks, Pete!

File: docs/README.md (hunk 1 above)
Comment (peterbrightwell) [reply docs/README.md]: PS the comment will appear in the final HTML.  There are ways of getting round that but they cause linting issues.  But I would expect spec authors to delete the comment anyway.

**Pull request merged**